sdk/spring/azure-spring-boot-test-selenium-common/src/main/java/com/azure/test/aad/common/SeleniumITHelper.java

```diff
@@ -35,6 +43,21 @@ protected void createDriver() {
         }
     }
 
+    private String getWebDriverCachePath() {
+        String currentPath = this.getClass().getResource(("")).getPath();
+        String sdkSpring = File.separator + "sdk" + File.separator + "spring";
+        String destination = currentPath + File.separator + WEB_DRIVER_FOLDER_NAME;
+        while (StringUtils.isNotEmpty(currentPath)) {
+            if (StringUtils.endsWith(currentPath, sdkSpring)) {
+                destination = currentPath + File.separator + WEB_DRIVER_FOLDER_NAME;
+                break;
+            } else {
+                currentPath = new File(currentPath).getParent();
+            }
+        }
+        return destination;
+    }
```

Comment:
This will make code easier to understand:
```
private String getWebDriverCachePath() {
    String springPath = this.getClass().getResource(("")).getPath();
    String springPathSuffix = File.separator + "sdk" + File.separator + "spring";
    while (StringUtils.isNotEmpty(springPath) && !springPath.endsWith(springPathSuffix)) {
        springPath = new File(springPath).getParent();
    }
    return springPath + File.separator + WEB_DRIVER_FOLDER_NAME;
}
```

Reply:
I am very happy to be working with so many genius